test/integrations/validation/Zod.spec.ts

```diff
@@ -0,0 +1,348 @@
+        });
+    });
+
+    describe('should validate using the Array schema', () =>
+    {
+        it('should validate roles', () =>
+        {
+            const input = { roles: ['admin', 'user'] };
+            zod.validate(input, SCHEMAS.ARRAY);
+        });
+
+        it('should require roles', () =>
+        {
+            const input = { roles: undefined };
+            const test = () => zod.validate(input, SCHEMAS.ARRAY);
+
+            expect(test).toThrowError(ERRORS.ARRAY);
+        });
+
+        it('should require roles to be an array', () =>
+        {
+            const input = { roles: 'admin' };
+            const test = () => zod.validate(input, SCHEMAS.ARRAY);
+
+            expect(test).toThrowError(ERRORS.ARRAY);
+        });
+
+        it('should require roles to be at least 1', () =>
+        {
+            const input = { roles: [] };
+            const test = () => zod.validate(input, SCHEMAS.ARRAY);
+
+            expect(test).toThrowError(ERRORS.ARRAY);
+        });
+
+        it('should require roles to be at most 2', () =>
+        {
+            const input = { roles: ['admin', 'user', 'guest'] };
+            const test = () => zod.validate(input, SCHEMAS.ARRAY);
+
+            expect(test).toThrowError(ERRORS.ARRAY);
+        });
+
+        it('should require roles to be at least 3 characters', () =>
+        {
+            const input = { roles: ['ad'] };
+            const test = () => zod.validate(input, SCHEMAS.ARRAY);
+
+            expect(test).toThrowError(ERRORS.ARRAY);
+        });
+
+        it('should require roles to be at most 5 characters', () =>
+        {
+            const input = { roles: ['admin', 'guestt'] };
+            const test = () => zod.validate(input, SCHEMAS.ARRAY);
+
+            expect(test).toThrowError(ERRORS.ARRAY);
+        });
+    });
+
+    describe('should not accept additional fields', () =>
+    {
+        it('should identify the additional fields', () =>
+        {
+            const input = { name: 'John', age: 10, length: 10 };
+            const test = () => zod.validate(input, SCHEMAS.STRING);
+
+            expect(test).toThrowError(ERRORS.ADDITIONAL_FIELDS);
+        });
+    });
+
+    describe('should validate using the Optional schema', () =>
+    {
+        it('should allow all missing fields', () =>
+        {
+            const input = {};
+            zod.validate(input, SCHEMAS.OPTIONAL);
+        });
+
+        it('should allow name to be missing', () =>
+        {
+            const input = { age: 18 };
+            zod.validate(input, SCHEMAS.OPTIONAL);
+        });
+
+        it('should allow age to be missing', () =>
+        {
+            const input = { name: 'John' };
+            zod.validate(input, SCHEMAS.OPTIONAL);
+        });
+    });
+
+    describe('should validate using the Creator schema', () =>
+    {
+        it('should validate creator', () =>
+        {
+            const input = {
```

Comment:
This looks like fixture data... Besides that, we try to avoid using domain concepts in our tests. If the domain changes, this could lead to confusion. Using unrelated concepts make it clear this data is purely related to the test.